crud.rb

```diff
@@ -46,19 +61,42 @@ def initialize
   erb :"users/edit"
 end
 
+# Login - Checks user credentials
+#-----------------------------------------------------------------
+post '/users/login' do
+  @user = @users.where(username: params[:username]).first
+  if @user == nil
+    erb :"homepage/error"
+  else
+    # password_salt = @user[:password_salt]
+    password = BCrypt::Engine.hash_secret(params[:password], @user[:password_salt])
+    if @user[:password_hash] == password
+      erb :"users/show"
```

Comment:
Before redirecting, you'll need to store the `@user`'s id into the session. You'll need to add `enable :sessions` line close the top of this file and then do something like:

```ruby
session[:current_user_id] = @user.id
```

To learn more about sessions in Sinatra, read this: 
http://www.sinatrarb.com/faq.html#sessions
https://en.wikipedia.org/wiki/Session_(computer_science)#Web_server_session_management - or about sessions in general